be/src/vec/functions/array/function_array_element.h

```diff
@@ -81,6 +94,79 @@ class FunctionArrayElement : public IFunction {
     }
 
 private:
+    //=========================== map element===========================//
+    ColumnPtr _get_mapped_idx(const ColumnArray& key_column,
+                           const ColumnWithTypeAndName& argument) {
+        if (key_column.get_data().is_column_string()) {
+            return _mapped_key_string(key_column, argument);
+        }
+        return nullptr;
```

Comment:
add TODO to support other key types

Reply:
using column.compare_at can cover other key types